ifrs17/Import/Importers.ipynb

```diff
@@ -503,18 +503,33 @@
         {
             "cell_type": "code",
             "source": [
-                "public async Task<ImportArgs[]> GetAllArgsAsync(ImportArgs args, IDataSource dataSource, string format, IDataSource targetDataSource)",
+                "public async Task<ImportArgs[]> GetAllArgsAsync(ImportArgs args, IDataSource dataSource, string format)",
                 "\n{",
                 "\n    ImportArgs[] allArgs;",
                 "\n    switch(format)",
                 "\n    {",
-                "\n        case ImportFormats.YieldCurve : ",
+                "\n        case ImportFormats.Cashflow:",
+                "\n        case ImportFormats.Actual: ",
+                "\n        case ImportFormats.Opening:",
+                "\n        case ImportFormats.SimpleValue:",
+                "\n        {",
+                "\n            if(args.Scenario != null) return args.RepeatOnce().ToArray();",
+                "\n            var secondaryArgs = await dataSource.Query<PartitionByReportingNodeAndPeriod>() ",
```

Comment:
Should it go the parsing storage and cached?

Reply:
Yes, I agree. If we used the parsing storage solely to query the database only once (per type), then we would recycle the data without querying multiple times. This is doable but it would require quite a few changes in the importers: both in the parsing storage to ensure that all needed types are queried, and also in these utility helper methods to be able to pass the parsing storage as parameter.

Reply:
I suggest to implement these changes in a separate PR. Can I?

Reply:
I created this ticket to this aim
https://github.com/Systemorph/IFRS17CalculationEngine/issues/237